app/forms/openid_connect_authorize_form.rb

```diff
@@ -81,7 +81,7 @@ def link_identity_to_service_provider(current_user, rails_session_id)
   end
 
   def success_redirect_uri
-    uri = redirect_uri unless errors.include?(:redirect_uri)
+    uri = redirect_uri unless (errors.include?(:redirect_uri) || errors.include?(:client_id))
```

Comment:
let's rewrite this? `unless` with a compound statement is really hard to follow IMO

```suggestion
    uri = redirect_uri if !(errors.include?(:redirect_uri) || errors.include?(:client_id))
```

Reply:
and we might even be able to turn this to an early return?

```suggestion
    return if errors.include?(:redirect_uri) || errors.include?(:client_id)
    uri = redirect_uri # probably remove this assignment and inline it below
```

Reply:
nice!